### PR TITLE
WFLY-8439 Unregistering node throws error: MEM: Can't update or inser…

### PR DESCRIPTION
**Pull request by stuartwdouglas**
…t context

### CODE REVIEW

**Comment by ctomc** [comment]
this is all wrong, especially around capabilities, it shouldn't have been merged ...
I will try to fix it properly as part of #9858  work.

**Comment by rhusar** [comment]
I have just stumbled upon this since I noticed that a new capability was added to ModClusterSubsystemResourceDefinition  but the same capability named "org.wildfly.mod_cluster" already exists at org.wildfly.extension.mod_cluster.ModClusterConfigResourceDefinition

**Comment by kabir** [comment]
I'll revert this

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java [hunk 1]
```diff
@@ -80,14 +86,22 @@ private HttpInvokerDefinition() {
...
     private static final class HttpInvokerAdd extends AbstractAddStepHandler {
 
         public HttpInvokerAdd() {
             super(ATTRIBUTES);
         }
 
         @Override
+        protected void recordCapabilitiesAndRequirements(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+            super.recordCapabilitiesAndRequirements(context, operation, resource);
+            final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
```
Comment (pferraro) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: OperationContext.getCurrentAddress()

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java [hunk 2]
```diff
@@ -80,14 +86,22 @@ private HttpInvokerDefinition() {
...
+            final PathAddress hostAddress = address.subAddress(0, address.size() - 1);
```
Comment (pferraro) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: address.getParent()

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java [hunk 3]
```diff
@@ -80,14 +86,22 @@ private HttpInvokerDefinition() {
...
+            final PathAddress serverAddress = hostAddress.subAddress(0, hostAddress.size() - 1);
```
Comment (pferraro) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: hostAddress.getParent()
:)

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java [hunk 4]
```diff
@@ -80,14 +86,22 @@ private HttpInvokerDefinition() {
...
+            context.registerCapability(RuntimeCapability.Builder.of(CAPABILITY_NAME + "." + serverAddress.getLastElement().getValue() + "." + hostAddress.getLastElement().getValue(), false).build());
```
Comment (pferraro) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: Shouldn't this registered with the ManagementResourceRegistration?

**Comment by ctomc** [comment]
I have integrated the code from this PR into https://github.com/wildfly/wildfly/pull/9858

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java (hunk 4 above)
Comment (ctomc) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: fixed in #9858

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java (hunk 3 above)
Comment (ctomc) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: fixed in #9858

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java (hunk 2 above)
Comment (ctomc) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: fixed in #9858

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java (hunk 1 above)
Comment (ctomc) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpInvokerDefinition.java]: fixed in #9858

**Pull request merged**